web/src/pages/contacts/index.tsx

```diff
@@ -1,125 +1,80 @@
 ﻿import ContentContainer from "../../components/design/ContentContainer";
-import InputText from "../../components/forms/InputText";
-import InputDate from "../../components/forms/InputDate";
 import Button from "../../components/forms/Button";
 import {NextPage} from "next";
 import {useTranslation} from "react-i18next";
-import {useForm} from "react-hook-form";
-import {useEffect, useState} from "react";
 import {useRouter} from "next/router";
 import Section from "../../components/design/Section";
 import {Contact} from "@/libs/models";
-import * as yup from "yup";
-import {yupResolver} from "@hookform/resolvers/yup";
-import useKeyboardJs from "react-use/lib/useKeyboardJs";
 import {searchContacts} from "@/libs/search";
+import {useMountOnce, useObservable} from "@legendapp/state/react";
+import ReactiveInputText from "@/components/forms/ReactiveInputText";
+import ReactiveInputDate from "@/components/forms/ReactiveInputDate";
+import { usePersistedObservable } from "@legendapp/state/react-hooks/usePersistedObservable"
+import { ObservablePersistSessionStorage } from '@legendapp/state/persist-plugins/local-storage'
 
 const Contacts: NextPage = () => {
     const { t } = useTranslation("common");
     const router = useRouter();
 
-    const [contacts, setContacts] = useState<Contact[] | null>(null);
-
-    const schema = yup.object({}).required();
-
-    const {
-        register,
-        handleSubmit,
-        formState: { errors },
-        setValue,
-        setFocus,
-        reset,
-    } = useForm({ resolver: yupResolver(schema) });
-
-    useEffect(() => {
-        const sessionSearchContacts = sessionStorage.getItem("contacts/search-contacts");
-        if (sessionSearchContacts) {
-            const data = JSON.parse(sessionSearchContacts);
-            Object.getOwnPropertyNames(data).forEach((field) => {
-                setValue(field, data[field]);
-            });
-            onSubmit(data);
+    const contacts$ = useObservable<Contact[] | null>(null);
+    const contacts = contacts$.use();
+    
+    const search$ = usePersistedObservable({
+        lastName: "",
+        firstName: "",
+        phoneNumber: "",
+        birthDate: "",
+    }, { persistLocal: ObservablePersistSessionStorage, local: "contacts/search-contacts" });
+    
+    useMountOnce(() => {
+        if (search$.lastName.get()?.length > 0 || search$.firstName.get()?.length > 0 || search$.phoneNumber.get()?.length > 0 || search$.birthDate.get()?.length > 0)
```

Comment:
Je suis pas sûr d'être fan du `?.length > 0`
pourquoi pas simplement:
```suggestion
        if (search$.lastName.get() || search$.firstName.get() || search$.phoneNumber.get() || search$.birthDate.get())
```